Assets/Mirror/Runtime/NetworkReader.cs

```diff
@@ -195,11 +195,8 @@ public byte[] ReadBytesAndSize()
 
         public ArraySegment<byte> ReadBytesAndSizeSegment()
         {
-            if (ReadBoolean())
-            {
-                return ReadBytesSegment((int)ReadPackedUInt32());
-            }
-            return default;
+            byte[] bytes = ReadBytesAndSize();
```

Comment:
This change completely defeats the purpose of ReadBytesAndSizeSegment(),  which was to read bytes in O(1) and without allocations